app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1354,15 +1399,72 @@ private void populatePlacesForAnotherLocation(
                 }));
     }
 
+    @SuppressLint("CheckResult")
+    public void loadPlacesDataAsync(List<Place> placeList, LatLng curLatLng) {
+        List<Place> places = new ArrayList<>(placeList);
```

Comment:
It would be great to sort the list here, in order of growing distance to curLatLng.